testing/python/transform/test_tilelang_transform_merge_shared_memory_allocations.py

```
@@ -0,0 +1,1040 @@
+    mod = tvm.IRModule({"main": func})
+    with tvm.transform.PassContext(config=config or {}):
+        mod = tilelang.transform.MergeSharedMemoryAllocations(enable_aggressive_merge=aggressive)(mod)
+    return mod["main"]
+
+
+def _load_prim_func_from_source(source: str, file_name: str, func_name: str = "before"):
+    from pathlib import Path
+
+    Path(file_name).write_text(source)
+    ns = {}
+    exec(compile(source, file_name, "exec"), ns)
```

Ruff (0.15.12)

[error] Use of `exec` detected

(S102)